website/blog/2022-07-05-date-trunc-sql-love-letter.md

```diff
@@ -0,0 +1,101 @@
+---
+title: "DATE_TRUNC SQL function: Why we love it"
+description: "The DATE_TRUNC function will truncate a date or time to the first instance for a given date part maintaining a date format. Wordy, wordy, wordy! What does this really mean?"
+slug: date-trunc-sql
+
+authors: [kira_furuichi]
+
+tags: [sql magic]
+hide_table_of_contents: false
+
+date: 2022-07-05
+is_featured: false
+---
+In general, data people prefer the more granular over the less granular. [Timestamps > dates](https://docs.getdbt.com/blog/when-backend-devs-spark-joy#signs-the-data-is-sparking-joy), daily data > weekly data, etc.; having data at a more granular level always allows you to zoom in. However, you’re likely looking at your data at a somewhat zoomed-out level—weekly, monthly, or even yearly. To do that, you’re going to need a handy dandy function that helps you round out date or time fields.
+
+The DATE_TRUNC function will truncate a date or time to the first instance for a given date part maintaining a date format. Wordy, wordy, wordy! What does this really mean? If you were to truncate ‘2021-12-13’ out to its month, it would return ‘2021-12-01’.
```

Comment:
```suggestion
The DATE_TRUNC function will truncate a date or time to the first instance of a given date part. Wordy, wordy, wordy! What does this really mean? If you were to truncate `2021-12-13` out to its month, it would return `2021-12-01` (the first day of the month).
```